tests/Microsoft.Data.DataFrame.Tests/ArrayComparer.cs

```diff
@@ -0,0 +1,169 @@
+        public void Visit(DoubleArray array) => CompareArrays(array);
+        public void Visit(BooleanArray array) => CompareArrays(array);
+        public void Visit(TimestampArray array) => CompareArrays(array);
+        public void Visit(Date32Array array) => CompareArrays(array);
+        public void Visit(Date64Array array) => CompareArrays(array);
+        public void Visit(ListArray array) => throw new NotImplementedException();
+        public void Visit(StringArray array) => CompareArrays(array);
+        public void Visit(BinaryArray array) => throw new NotImplementedException();
+        public void Visit(IArrowArray array) => throw new NotImplementedException();
+
+        private void CompareArrays<T>(PrimitiveArray<T> actualArray)
+            where T : struct, IEquatable<T>
+        {
+            Assert.IsAssignableFrom<PrimitiveArray<T>>(_expectedArray);
+            PrimitiveArray<T> expectedArray = (PrimitiveArray<T>)_expectedArray;
+
+            Assert.Equal(expectedArray.Length, actualArray.Length);
+            Assert.Equal(expectedArray.NullCount, actualArray.NullCount);
+            Assert.Equal(expectedArray.Offset, actualArray.Offset);
+
+            Assert.True(expectedArray.NullBitmapBuffer.Span.SequenceEqual(actualArray.NullBitmapBuffer.Span));
+            Assert.True(expectedArray.Values.Slice(0, expectedArray.Length).SequenceEqual(actualArray.Values.Slice(0, actualArray.Length)));
+        }
+
+        private void CompareArrays(BooleanArray actualArray)
+        {
+            Assert.IsAssignableFrom<BooleanArray>(_expectedArray);
+            BooleanArray expectedArray = (BooleanArray)_expectedArray;
+
+            Assert.Equal(expectedArray.Length, actualArray.Length);
+            Assert.Equal(expectedArray.NullCount, actualArray.NullCount);
+            Assert.Equal(expectedArray.Offset, actualArray.Offset);
+
+            Assert.True(expectedArray.NullBitmapBuffer.Span.SequenceEqual(actualArray.NullBitmapBuffer.Span));
+            int booleanByteCount = BitUtility.ByteCount(expectedArray.Length);
+            Assert.True(expectedArray.Values.Slice(0, booleanByteCount).SequenceEqual(actualArray.Values.Slice(0, booleanByteCount)));
+        }
+
+        private void CompareArrays(StringArray actualArray)
+        {
+            Assert.IsAssignableFrom<StringArray>(_expectedArray);
+            StringArray expectedArray = (StringArray)_expectedArray;
+
+            Assert.Equal(expectedArray.Length, actualArray.Length);
+            Assert.Equal(expectedArray.NullCount, actualArray.NullCount);
+            Assert.Equal(expectedArray.Offset, actualArray.Offset);
+
+            Assert.True(expectedArray.NullBitmapBuffer.Span.SequenceEqual(actualArray.NullBitmapBuffer.Span));
+            Assert.True(expectedArray.Values.Slice(0, expectedArray.Length).SequenceEqual(actualArray.Values.Slice(0, actualArray.Length)));
+        }
+    }
+
+    internal class FieldComparer
```

Comment:
This and the classes below should all be `static` classes since they only have static methods.